lib/index.js

```diff
@@ -1,16 +1,31 @@
-const { compress: _compress, decompress: _decompress } = require('bindings')('zstd');
+'use strict';
+const zstd = require('bindings')('zstd');
+const { promisify } = require('util');
 
+const _compress = promisify(zstd.compress);
+const _decompress = promisify(zstd.decompress);
 // Error objects created via napi don't have JS stacks; wrap them so .stack is present
 // https://github.com/nodejs/node/issues/25318#issuecomment-451068073
 
-exports.compress = async function compress(data) {
+exports.compress = async function compress(data, compressionLevel) {
+  if (!Buffer.isBuffer(data)) {
+    throw new TypeError(`parameter 'data' must be a Buffer.`);
+  }
+
+  if (compressionLevel != null && typeof compressionLevel !== 'number') {
+    throw new TypeError(`parameter 'compressionLevel' must be a number.`);
+  }
+
   try {
-    return await _compress(data);
+    return await _compress(data, compressionLevel ?? 3);
   } catch (e) {
     throw new Error(`zstd: ${e.message}`);
   }
 };
 exports.decompress = async function decompress(data) {
+  if (!Buffer.isBuffer(data)) {
+    throw new TypeError(`parameter 'data' must be a Buffer.`);
+  }
```

Comment:
1.x supports Uint8Array, why are we removing support for that?

Reply:
yeah, good catch.  I didn't look closely enough at the actual implementation.  our types declare Buffers specifically